tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleNameMojo.java

```diff
@@ -0,0 +1,128 @@
+        String[] words = kebabCase.split("-");
+        StringBuilder result = new StringBuilder();
+
+        // Capitalize first letter of each word
+        for (int i = 0; i < words.length; i++) {
+            String word = words[i];
+            if (!word.isEmpty()) {
+                result.append(Character.toUpperCase(word.charAt(0)));
+                if (word.length() > 1) {
+                    result.append(word.substring(1).toLowerCase());
+                }
+                // Only add %20 if it's not the last word
+                if (i < words.length - 1) {
+                    result.append("%20");
+                }
+            }
+        }
+
+        return result.toString();
+    }
+
+    @Override
+    protected boolean processWrappedBundle(WrappedBundle wrappedBundle) {
+        Bundle bundle = wrappedBundle.getBundle();
+        String location = bundle.getLocation();
+        try {
+            bundle.setLocation(processLocation(wrappedBundle));
+        } catch (Exception e) {
+            getLog().error("Could not process the Bundle location '%s': %s".formatted(location, e.getMessage()), e);
+        }
+        return false;
+    }
+
+    String processLocation(WrappedBundle wrappedBundle) throws Exception {
+        String location = wrappedBundle.getBundle().getLocation();
+        String instructions = wrappedBundle.getInstructions();
+        AtomicBoolean dollarNeeded = new AtomicBoolean(!(instructions!=null && instructions.contains("$")));
+
+        String bundleNameHeader = "%s=Wrap%%20of%%20%s".formatted(BUNDLE_NAME, toPascalCase(wrappedBundle.getArtifactId()));
+        String bundleSymbolicNameHeader = "%s=wrap_%s.%s".formatted(BUNDLE_SYMBOLIC_NAME, wrappedBundle.getGroupId(), wrappedBundle.getArtifactId());
+
+        location = insertHeaderIfNeeded(dollarNeeded, location, BUNDLE_NAME, bundleNameHeader);
+        return insertHeaderIfNeeded(dollarNeeded, location, BUNDLE_SYMBOLIC_NAME, bundleSymbolicNameHeader);
+    }
+
+    private String insertHeaderIfNeeded(AtomicBoolean dollarNeeded, String location, String headerName, String fullHeader) {
+        if (location.contains(headerName)) {
+            return location;
+        }
+        StringBuilder sb = new StringBuilder(location);
+
+        // insert before existing headers header
+        for (String header : HEADERS_AFTER_BUNDLE_NAME) {
+            // add Bundle-Version before
+            if (location.contains(header)) {
+                int versionHeaderStartIndex = location.indexOf(header);
+                if (!dollarNeeded.get()) {
+                    // "amp;" is automatically added
+                    return sb.insert(versionHeaderStartIndex, "%s&".formatted(fullHeader)).toString();
+                } else {
+                    // "amp;" is automatically added
+                    dollarNeeded.set(false);
+                    return sb.insert(versionHeaderStartIndex, "$%s&".formatted(fullHeader)).toString();
+                }
```

Comment:
For a better readibility, I would keep the same order as below https://github.com/apache/camel-karaf/pull/591/files#diff-ab619fa88de63d711e3925a32a40a68831b014167fe830aad41c777daa7d823aR120-R126. I mean if dollarNeeded.get() ... else